fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/models/ResourceFactory.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.api.models;
+
+/**
+ * Interface to a factory to locate or create new resources.
+ *
+ * @author whikloj
+ * @since 2019-09-23
+ */
+public interface ResourceFactory {
```

Comment:
what would you all think of including a `public FedoraResource getResource(identifier)` method here. It would throw an error if a resource wasn't found rather than initializing a new object. The object returned would be of the appropriate subclass, but it would be returned as a a FedoraResource, so callers would have to cast it if they cared about the difference. This would be used for GET and HEAD cases

Reply:
I am fine with this and added something like this as `private` to the `ResourceFactoryImpl` stub.

Reply:
+1